apps/api/src/v1/workspaces/workspace/documents/document/settings.ts

```diff
@@ -0,0 +1,56 @@
+import { Router } from 'express'
+import { getParam } from '../../../../../utils/express.js'
+import { z } from 'zod'
+import { IOServer } from '../../../../../websocket/index.js'
+import prisma from '@briefer/database'
+import { broadcastDocument } from '../../../../../websocket/workspace/documents.js'
+
+const DocumentSettings = z.object({
+  runUnexecutedBlocks: z.boolean(),
+})
+
+export default function settingsRouter(socketServer: IOServer) {
+  const publishRouter = Router({ mergeParams: true })
+
+  publishRouter.put('/', async (req, res) => {
+    const workspaceId = getParam(req, 'workspaceId')
+    const documentId = getParam(req, 'documentId')
+    const body = DocumentSettings.safeParse(req.body)
+    if (!body.success) {
+      res.status(400).json({ reason: 'invalid-payload' })
+      return
+    }
+
+    const runUnexecutedBlocks = body.data.runUnexecutedBlocks
+
+    try {
+      await prisma().$transaction(async (tx) => {
+        const doc = await tx.document.update({
+          where: { id: documentId },
+          data: { runUnexecutedBlocks },
+        })
+
+        if (!doc) {
+          res.status(404).end()
+          return
+        }
+      })
```

Comment:
This transaction is not needed, also that early return will not prevent further execution of request handling it just returns from the transaction callback.
Furthermore, `update` does not return `null` it will crash if row is not found.
You should remove the transaction, catch the not found error and send 404

Reply:
You can also use the `recoverFromNotFound` helper, that will transform the not found error into null.